oxy_fitting_ver2.py

```diff
@@ -0,0 +1,447 @@
+        plt.plot(time_data_matched['Fitted_OXY_uMOLKG_time'],time_data_matched['CTDPRS_y'],'g',label='time data')
+        plt.legend()
+        plt.xlabel('Oxygen (uMol/KG)')
+        plt.ylabel('Pressure (dBar)')
+        plt.show()
+
+        btl_data_clean['OXYGEN'] = btl_data_clean['OXY_uMOLKG_btl']
+        btl_data_clean['CTDOXY'] = time_data_matched['Fitted_OXY_uMOLKG_time']
+
+        btl_data_output = btl_data_clean
+        btl_data_clean['residual'] = np.abs(btl_data_output['OXYGEN']-btl_data_output['CTDOXY'])
+        std_res = np.std(btl_data_output['residual'])
+        cutoff = np.std(btl_data_output['residual'])*2.8
+        
+        btl_data_clean=btl_data_clean[btl_data_output['residual']<=cutoff]
+        bad_values2 = btl_data_output[btl_data_output['residual']>cutoff]
+        thrown_values = btl_data_clean[btl_data_output['residual']>cutoff]
+        bad_values = pd.concat([bad_values,bad_values2])
+        print('NEW STANDARD DEVIATION IS:',std_res)
+    #Add Station and Cast Numbers to DataFrame
+    time_data_matched['STNNBR']=stn_nbr
+    time_data_matched['CASTNO']=cst_nbr
+
+    btl_data_clean['STNNBR']=stn_nbr
+    btl_data_clean['CASTNO']=cst_nbr
+    
+    plt.plot(btl_data_clean['residual'],btl_data_clean['CTDPRS']*-1,'bx')
+    plt.xlim(xmax=10)
+    plt.show()
+    
+    bad_values['CTDOXY_FLAG_W'] = 4
+    btl_data_clean['CTDOXY_FLAG_W'] = 2
+    btl_data_clean = pd.concat([btl_data_clean,bad_values])
+    btl_data_clean=btl_data_clean.sort_values(by='BTLNBR')
+    
+    btl_data_clean = btl_data_clean.reset_index(drop=True)
+    btl_data_write = pd.DataFrame()
+    btl_data_write['STNNBR'] = btl_data_clean['STNNBR'].values
+    btl_data_write['CASTNO'] = btl_data_clean['CASTNO'].values
+    btl_data_write['SAMPNO'] = btl_data_clean['BTLNBR'].values
+    btl_data_write['CTDOXY'] = btl_data_clean['CTDOXY'].values
+    btl_data_write['CTDOXY_FLAG_W'] = btl_data_clean['CTDOXY_FLAG_W'].values
+    
+    return btl_data_write, btl_data_clean
+
+def oxygen_cal_ml(coef0,time_data,btl_data,switch):
+
+    """"
+
+    NOAA's oxygen fitting routine using the equation:
+    OXY(ml/L) = SOC * (doxy_volts+Voffset+Tau20*exp(cc1*PRES+cc2*TEMP)*dvdt) \
+                *os*exp(Tcorr*TEMP)*exp(E*PRESS/TEMP_K)
+
+    coef0s:
+    coef[0] = Soc
+    coef[1] = Voffset
+    coef[2] = Tau20
+    coef[3] = Tcorr
+    coef[4] = E
+
+    cc[0] = D1
+    cc[1] = D2
+    
+    """
+
+    cc = [1.92634e-4,-4.64803e-2]
+    #ORIGINAL CODE
+    #time_data['NOAA_oxy_mlL'] =coef0[0]*(time_data['CTDOXYVOLTS_y']+\
+    #coef0[1]+coef0[2]*np.exp(cc[0]*time_data['CTDPRS_y']+cc[0]*time_data['TEMPERATURE_CTD'])\
+    #*time_data['dv_dt_time'])*time_data['OS']*np.exp(coef0[3]*time_data['TEMPERATURE_CTD'])\
+    #*np.exp((coef0[4]*time_data['CTDPRS_y'])/(time_data['TEMPERATURE_CTD']+273.15))
+
+    #MODIFIED CODE
+    time_data['NOAA_oxy_mlL'] =coef0[0]*(time_data['CTDOXYVOLTS_time']+\
```

Comment:
https://www.python.org/dev/peps/pep-0008/#should-a-line-break-before-or-after-a-binary-operator